src/Http/Http.Extensions/src/RequestDelegateFactory.cs

```diff
@@ -466,24 +468,22 @@ private static Expression CreateRouteHandlerInvocationContextBase(FactoryContext
             _ => typeof(DefaultRouteHandlerInvocationContext)
         };
 
-        if (constructorType.IsGenericType)
+        if (!RuntimeFeature.IsDynamicCodeCompiled || !constructorType.IsGenericType)
```

Comment:
I wonder if we should remove all of these checks now that we're going full source generator.

Reply:
@halter73 Sounds like this needs to be updated after the introduction of RDG.  @captainsafia might have thoughts?